src/persistence/profile.cpp

```diff
@@ -520,12 +525,19 @@ void Profile::setPassword(QString newPassword)
 {
     QList<HistoryKeeper::HistMessage> oldMessages = HistoryKeeper::exportMessagesDeleteFile();
     QByteArray avatar = loadAvatarData(core->getSelfId().publicKey);
-
+    QString oldPassword = password;
     password = newPassword;
     passkey = *core->createPasskey(password);
     saveToxSave();
-
     HistoryKeeper::getInstance()->importMessages(oldMessages);
     Nexus::getDesktopGUI()->reloadHistory();
     saveAvatar(avatar, core->getSelfId().publicKey);
+
+    QVector<uint32_t> friendList = core->getFriendList();
+    QVectorIterator<uint32_t> i(friendList);
+    while (i.hasNext())
+    {
+        QString friendPublicKey = core->getFriendPublicKey(i.next());
+        saveAvatar(loadAvatarData(friendPublicKey,oldPassword),friendPublicKey);
```

Comment:
Currently loadAvatarData calls avatarPath without giving it a password, so it uses the profile password, and this happens *after* setting the new profile password.
So if we used to have no password and we just set one, instead of loading the old unencrypted avatars avatarPath will see the new profile password and return the encrypted path.